### PR TITLE
Misc Code Cleanups

### PR DESCRIPTION
**Pull request by res0nance**
Signed-off-by: Raihaan Shouhell <raihaan.shouhell@autodesk.com>

Doing some misc code cleanups

### Proposed changelog entries

* Not needed

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### CODE REVIEW

File: test/src/test/java/jenkins/security/stapler/DoActionFilterTest.java [hunk 1]
```diff
@@ -350,7 +350,7 @@ public void validateCrumb(StaplerRequest request, String submittedCrumb) {
         WebRequest settings = new WebRequest(new URL(j.getURL(), "testNewRulesOk/annotatedJavascriptScriptMethod/"));
         settings.setAdditionalHeader("Content-Type", "application/x-stapler-method-invocation");
         settings.setHttpMethod(HttpMethod.POST);
-        settings.setRequestBody(JSONArray.fromObject(Arrays.asList()).toString());
+        settings.setRequestBody(JSONArray.fromObject(Collections.emptyList()).toString());
```
Comment (zbynek) [inline test/src/test/java/jenkins/security/stapler/DoActionFilterTest.java]: Not sure if this is really more readable than
```
settings.setRequestBody("[]");
```

File: core/src/main/java/jenkins/util/xstream/XStreamDOM.java [hunk 2]
```diff
@@ -308,8 +308,10 @@ public String xpath() {
                         count++;
                 boolean more = false;
                 for (int i=pos; !more && i<node.children.size(); i++)
-                    if (node.children.get(i).tagName.equals(child.tagName))
+                    if (node.children.get(i).tagName.equals(child.tagName)) {
                         more = true;
+                        break;
```
Comment (alecharp) [inline core/src/main/java/jenkins/util/xstream/XStreamDOM.java]: The `more` is used in the `for` loop condition, so no need to scare the static code analysis :wink:

File: cli/src/main/java/hudson/cli/CLI.java [hunk 3]
```diff
@@ -39,10 +39,7 @@
 import java.security.GeneralSecurityException;
 import java.security.KeyPair;
 import java.security.SecureRandom;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Locale;
-import java.util.Properties;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (alecharp) [reply cli/src/main/java/hudson/cli/CLI.java]: -1 on `*` imports.

File: core/src/test/java/hudson/cli/ListJobsCommandTest.java [hunk 4]
```diff
@@ -14,10 +14,7 @@
 
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
+import java.util.*;
```
Comment (varyvol) [inline core/src/test/java/hudson/cli/ListJobsCommandTest.java]: Even if this test code, probably discouraged to use star imports.

File: test/src/test/java/hudson/model/FingerprintTest.java [hunk 5]
```diff
@@ -35,12 +35,7 @@
 
 import java.io.File;
 import java.io.IOException;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Hashtable;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (varyvol) [inline test/src/test/java/hudson/model/FingerprintTest.java]: Ditto.

File: test/src/test/java/hudson/model/ViewTest.java [hunk 6]
```diff
@@ -60,11 +60,8 @@
 import hudson.util.HudsonIsLoading;
 import java.io.File;
 import java.io.IOException;
-import java.util.Collection;
+import java.util.*;
```
Comment (varyvol) [inline test/src/test/java/hudson/model/ViewTest.java]: Ditto.

File: test/src/test/java/hudson/model/queue/LoadPredictorTest.java [hunk 7]
```diff
@@ -44,10 +44,7 @@
 import org.jvnet.hudson.test.TestExtension;
 
 import java.lang.reflect.Field;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.GregorianCalendar;
-import java.util.List;
+import java.util.*;
```
Comment (varyvol) [inline test/src/test/java/hudson/model/queue/LoadPredictorTest.java]: Ditto.

File: core/src/test/java/hudson/cli/ListJobsCommandTest.java (hunk 4 above)
Comment (alecharp) [inline core/src/test/java/hudson/cli/ListJobsCommandTest.java]: same -1 on `*` import.

File: test/src/test/java/jenkins/security/stapler/DoActionFilterTest.java (hunk 1 above)
Comment (varyvol) [reply test/src/test/java/jenkins/security/stapler/DoActionFilterTest.java]: I would prefer this one as well.

File: test/src/test/java/hudson/model/FingerprintTest.java (hunk 5 above)
Comment (alecharp) [inline test/src/test/java/hudson/model/FingerprintTest.java]: `*` import dito

File: test/src/test/java/hudson/model/ViewTest.java (hunk 6 above)
Comment (alecharp) [inline test/src/test/java/hudson/model/ViewTest.java]: -1 `*`

File: test/src/test/java/hudson/model/queue/LoadPredictorTest.java (hunk 7 above)
Comment (alecharp) [inline test/src/test/java/hudson/model/queue/LoadPredictorTest.java]: -1 `*`

File: test/src/test/java/hudson/tasks/FingerprinterTest.java [hunk 8]
```diff
@@ -38,10 +38,7 @@
 import java.io.File;
 
 import java.io.IOException;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
+import java.util.*;
```
Comment (alecharp) [inline test/src/test/java/hudson/tasks/FingerprinterTest.java]: -1 `*`

**Review by alecharp — CHANGES_REQUESTED** [review]
-1 on `*` import for `java.util` package.

File: cli/src/main/java/hudson/cli/CLI.java [hunk 9]
```diff
@@ -40,15 +43,16 @@
...
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
 import java.util.Properties;
 import java.util.logging.Handler;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-import static java.util.logging.Level.*;
-import org.apache.commons.io.FileUtils;
```
Comment (MarkEWaite) [inline cli/src/main/java/hudson/cli/CLI.java]: Moving FileUtils and StringUtils earlier in the file seems like it doesn't change much and may cause conflicts with other pull requests.  Could those be left in their current location?

**Review by MarkEWaite — COMMENTED** [review]
I don't see the benefit of arranging import statements when there isn't an enforced policy on the order of import statements and rearranging them may cause conflicts in other pending pull requests.

Others with more experience may guide differently, so don't take this as a negative as much as a comment.  I don't understand what we gain.  Can you help me understand?

**Comment by res0nance** [comment]
@MarkEWaite There is no benefit, IDEs just do its magic and it gets ordered this way.

**Pull request closed without merging**